packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -66,26 +67,39 @@ const ProductTile = (props) => {
         ...rest
     } = props
 
-    const {currency, image, price, productId, hitType} = product
-
+    const {image, productId} = product
     // ProductTile is used by two components, RecommendedProducts and ProductList.
     // RecommendedProducts provides a localized product name as `name` and non-localized product
     // name as `productName`. ProductList provides a localized name as `productName` and does not
     // use the `name` property.
     const localizedProductName = product.name ?? product.productName
 
-    const {currency: activeCurrency} = useCurrency()
     const isFavouriteLoading = useRef(false)
     const styles = useMultiStyleConfig('ProductTile')
+    // NOTE: swatches will implement later to set variant accordingly,
+    // On first load, for variant product, get the variant that is the represent product
+    // Also, product tile can be used in RecommendedProducts where it calls getProducts which does not have representedProduct
+    // in that case we use the first variant that has price book to set up discount price
+    // Not all variants has set in a priceBook, meaning not having tieredPrices.
+    const variant = useMemo(
+        () =>
+            product?.variants?.find(
+                (i) => i?.productId === product?.representedProduct?.id || !!i?.tieredPrices
+            ),
+        [product]
+    )
 
+    const {listPrice, currentPrice} = getDisplayPrice({...product, ...variant})
+    const isASet = product?.hitType === 'set' || !!product?.type?.set
+    let productUrl = variant
```

Comment:
For consistency, for a variant product since product tile shows representedProduct price on PLP, we should link the users to that product when they click on the tile for variant.